src/Models/Scopes/Product/WithProductChildrenScope.php

```diff
@@ -39,10 +40,18 @@ public function apply(Builder $builder, Model $model)
                     "in_stock", children_stock.is_in_stock,
                     {$stockQty}
                     "images", (
-                        SELECT JSON_ARRAYAGG(catalog_product_entity_media_gallery.value)
-                        FROM catalog_product_entity_media_gallery_value_to_entity
-                        LEFT JOIN catalog_product_entity_media_gallery ON catalog_product_entity_media_gallery.value_id = catalog_product_entity_media_gallery_value_to_entity.value_id
-                        WHERE catalog_product_entity_media_gallery_value_to_entity.entity_id = children.entity_id
+                        SELECT JSON_ARRAYAGG(JSON_OBJECT("value", catalog_product_entity_media_gallery.value, "position", catalog_product_entity_media_gallery_value.position))
```

Comment:
JSON_ARRAYAGG and this subquery does not support an `ORDER BY`, thus sorting is done during the cast.